Add some things to make to make the widget more flexible

Hannnes1 opened the pull request:
Some of these things (like `isThreeLine`) comes directly from ´ListTile´, and some of the things I had to add to make it work in one of our apps.

You don't have to merge this, but I thought I would open a pull request anyway since I needed these changes anyway.

Hannnes1 committed:
Add initial elevation to constructor

Hannnes1 committed:
Add support for three-line listtile

Hannnes1 committed:
Rotate trailing widget even if it's custom

Hannnes1 committed:
Clean up a bit

Hannnes1 committed:
Add support for shadow color

Hannnes1 committed:
Add a color property that overwrites other colors

I couldn't get the other colors to work, and I feelt that this is
simpler if you want the same color all the time

Hannnes1 committed:
Revert "Add a color property that overwrites other colors"

This reverts commit 287087c9e599b262b28ee8beb72f9634f34e21cf.

Hannnes1 committed:
Revert "Add an elevation that overwrites all others"

This reverts commit f9ff756774d121f2d23b9e47142f5b89b3e0d8c9.

Hannnes1 committed:
Add an elevation that overwrites all others

Skylled commented on lib/expansion_tile_card.dart:
```diff
@@ -95,64 +102,77 @@ class ExpansionTileCard extends StatefulWidget {
   /// Defaults to 2.0.
   final double elevation;
 
+  ///The elevetion when collapsed
```
```suggestion
  /// The elevation when collapsed
```
Nit: Typo

Nit: Space before comments

Skylled commented on lib/expansion_tile_card.dart:
```diff
@@ -95,64 +102,77 @@ class ExpansionTileCard extends StatefulWidget {
 
+  ///The elevetion when collapsed
+  ///
+  ///Defaults to 0.0
```
```suggestion
  /// Defaults to 0.0
```

Skylled commented on lib/expansion_tile_card.dart:
```diff
@@ -254,13 +272,14 @@ class _ExpansionTileCardState extends State<ExpansionTileCard> with SingleTicker
                       contentPadding: widget.contentPadding,
                       leading: widget.leading,
                       title: widget.title,
                       subtitle: widget.subtitle,
-                      trailing: widget.trailing ?? RotationTransition(
+                      trailing: RotationTransition(
                         turns: _iconTurns,
-                        child: const Icon(Icons.expand_more),
+                        child: widget.trailing ?? Icon(Icons.expand_more),
```
This would be an unexpected change in behavior from the original `ExpansionTile` that this widget is based on. I'd like to mirror the original Flutter version as much as possible.

Hannnes1 replied:
My thought was that you wouldn't have to do the RotationTransition yourself if it was this way, but I see what you mean. Do you think I should just revert this commit?

Skylled replied:
Probably. Another option, since I could see why developers may want this, would be to have the rotation disabled by default if `trailing` is set and enabled with a new boolean. Not sure how well that would work though.

Skylled commented on lib/expansion_tile_card.dart:
```diff
@@ -95,64 +102,77 @@ class ExpansionTileCard extends StatefulWidget {
+  ///Defaults to 0.0
+  final double initialElevation;
+
+  final Color shadowColor;
```
Describe the variable in a comment.

Skylled commented on lib/expansion_tile_card.dart:
```diff
@@ -95,64 +102,77 @@ class ExpansionTileCard extends StatefulWidget {
   /// Defaults to Curves.easeIn.
   final Curve paddingCurve;
 
+  ///This color will overwrite all other colors
+  final Color color;
+
```
I'm not sure this is a good idea, or at the very least not with this name. At a minimum this should be renamed `overrideColor`, but I'm not sure why it's needed.

This job can already be done by setting `baseColor` and `expandedColor` to be the same, only difference being (possibly?) performance as you're skipping the animation.

Hannnes1 replied:
The reason I did this was because the color wouldn't change at all when I tried setting baseColor and expandedColor to the same thing. You are probably right that this is unnecessary and I should try to find the underlying cause instead.

Skylled replied:
Fair enough. I had a great deal of difficulty getting the colors to work as I expected them to. The issue may be something on Flutter's end. I'm fine accepting this with the name `overrideColor`

Skylled replied:
Or `constantColor` to match elevation

Hannnes1 replied:
So I figured this out... I didn't realize that the changes weren't applied until I restarted the app since they are set in initState. I have reverted this commit, as well as the one that added the constantElevation.

Skylled commented on lib/expansion_tile_card.dart:
```diff
@@ -95,64 +102,77 @@ class ExpansionTileCard extends StatefulWidget {
+  ///This color will overwrite all other colors
+  final Color color;
+
+  ///This elevation will overwrite all other
+  final double constantElevation;
+
```
Same here. I'm not sure why we need an override for elevation if you can just set `initialElevation` and `elevation` to the same value.

Hannnes1 replied:
Same as the color

Skylled replied:
That's...really interesting. I'll have to give it a try to see why that would be.

Skylled reviewed (changes requested):
Thanks for submitting this!

I have no problem with some of the changes here — `initialElevation`, `isThreeLine`, and `shadowColor` — to be precise. If you wanted to separate those into a different PR, I'd accept that no problem.

I'd like to hear more about the use cases for `color` and `constantElevation` though.

Hannnes1 committed:
Update lib/expansion_tile_card.dart

Co-authored-by: Kyle Bradshaw <TehSkull@hotmail.com>

Hannnes1 committed:
Fix typo

Co-authored-by: Kyle Bradshaw <TehSkull@hotmail.com>

Hannnes1 committed:
Add documentation to shadowColor property

Hannnes1 committed:
Don't animate custom trailing unless specified

If a trailing widget is specified, and the parameter `animateTrailing`
isn't true, an `AlwaysStoppedAnimation` will be shown instead.

Skylled commented:
Looks good! Thanks for contributing! I'll see about rolling a new version with these changes this week.
